lib/mshoplib/src/MShop/Plugin/Provider/Order/Coupon.php

```diff
@@ -79,7 +79,14 @@ public function update( \Aimeos\MW\Observer\Publisher\Iface $order, $action, $va
 			if( ( $couponItem = reset( $results ) ) !== false )
 			{
 				$couponProvider = $couponManager->getProvider( $couponItem, strtolower( $code ) );
-				$couponProvider->updateCoupon( $order );
+				try 
+				{
+					$couponProvider->updateCoupon( $order );
+				}
+				catch( \Exception $e )
+				{
+					$notAvailable[$code] = 'coupon.gone';
+				}
```

Comment:
This should be:
```
try {
	$couponProvider->updateCoupon( $order );
} catch( ( \Exception $e ) {
	$notAvailable[$code] = 'gone';
}
```
because `coupon.` is added in the HTML client like for products

Reply:
What about L93 - should I remove the `coupon.`, too? This might break existing installations...